t/t1091-sparse-checkout-builtin.sh

```diff
@@ -739,4 +748,34 @@ test_expect_success 'add with cone mode verifies existing cone patterns' '
 	test_i18ngrep "unable to use existing sparse-checkout patterns in cone mode" err
 '
 
+# NEEDSWORK: in the case of directory patterns like `deep/`, it might be worth trying
+# to "correct" the patterns to match a cone mode style. However, that may be more difficult
+# for nested directories (like `deep/deeper1/`) in which multiple individual patterns
+# would be mapped from the original (`/deep/`, `!/deep/*/`, `/deep/deeper1/`).
```

Comment:
This is an interesting idea, but I agree that the complexity might not be worth it.

I think "deep/" also means "include anything that's in a `deep/` directory, including subdirectories such as `/bin/deep/`", so we can't really convert in this way without changing the definition. Even scanning the index for these `deep/` directories is only enough for `HEAD` and not enough as `HEAD` moves throughout history.